services/ingestion/etl/agri/csv/src/lib/transform.js

```diff
@@ -40,51 +40,72 @@ export default (record: Object): Project => {
   };
 
   // Preprocess coordinators
-  const coordArray = record.Coordinators.split(';').map(coordinator => ({
-    name: coordinator,
-    type: '',
-    address: '',
-    region: '',
-    country: '',
-    website: '',
-    phone: '',
-    email: '',
-  }));
+  const coordArray = record.Coordinators.split(';')
+    .filter(coordinator => coordinator)
+    .map(coordinator => ({
+      name: coordinator,
+      type: '',
+      address: '',
+      region: '',
+      country: '',
+      website: '',
+      phone: '',
+      email: '',
+    }));
 
   // Preprocess partners
-  const partnerArray = record.Partners.split(',').map(partner => ({
-    name: partner,
-    type: '',
-    address: '',
-    region: '',
-    country: '',
-    website: '',
-  }));
+  const partnerArray = record.Partners.split(',')
+    .filter(partner => partner)
+    .map(partner => ({
+      name: partner,
+      type: '',
+      address: '',
+      region: '',
+      country: '',
+      website: '',
+    }));
 
   // Preprocess locations
   const longArray = record['Project location longitude'].split(';');
   const latArray = record['Project location latitude'].split(';');
   const locationArray = record['Project country(ies)']
     .split(';')
-    .map((country, index) => ({
-      country_code: country,
-      region: '',
-      nuts2: '',
-      address: record['Project address(es)'] || '',
-      postal_code: record['Project postal code(s)'] || '',
-      town: record['Project town(s)'] || '',
-      location: {
-        type: 'Point',
-        coordinates: [
-          parseFloat(Array.isArray(longArray) && longArray[index]) || 0,
-          parseFloat(Array.isArray(latArray) && latArray[index]) || 0,
-        ],
-      },
-    }));
+    .map((country, index) => {
```

Comment:
Some checks are repeated, maybe something like

```
  const hasLocations = Array.isArray(longArray) && Array.isArray(latArray);

  const locationArray = hasLocations
    ? record['Project country(ies)'].split(';').map((country, index) => {
        const hasCoordinates = longArray[index] && latArray[index];

        const latitude = parseFloat(latArray[index]) || 0;
        const longitude = parseFloat(longArray[index]) || 0;

        return {
          country_code: country,
          region: '',
          nuts2: '',
          address: record['Project address(es)'] || '',
          postal_code: record['Project postal code(s)'] || '',
          town: record['Project town(s)'] || '',
          centroid: hasCoordinates
            ? {
                lat: latitude,
                lon: longitude,
              }
            : null,
          location: hasCoordinates
            ? {
                type: 'Point',
                coordinates: [longitude, latitude],
              }
            : null,
        };
      })
    : [];
```

Would be a bit more readable

Reply:
In your case, `hasLocations` will ALWAYS be true, I don't see the point of the condition. Also, you assign `latitude` and `longitude` even if there's no coordinate, forcing things like `[0, 0]` when `null` would be a better fit